devicetypes/smartthings/ezex-temp-humidity-sensor.src/ezex-temp-humidity-sensor.groovy

```diff
@@ -0,0 +1,129 @@
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+			}
+		}
+		valueTile("humidity", "device.humidity", inactiveLabel: false, width: 2, height: 2) {
+			state "humidity", label: '${currentValue}% humidity', unit: ""
+		}
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery'
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "temperature", "humidity"
+		details(["temperature", "humidity", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	log.debug "description: $description"
+
+	// getEvent will handle temperature and humidity
+	Map map = zigbee.getEvent(description)
+	if (!map) {
+		Map descMap = zigbee.parseDescriptionAsMap(description)
+		if (descMap?.clusterInt == zigbee.TEMPERATURE_MEASUREMENT_CLUSTER && descMap.commandInt == 0x07) {
+			if (descMap.data[0] == "00") {
+				log.debug "TEMP REPORTING CONFIG RESPONSE: $descMap"
+				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+			} else {
+				log.warn "TEMP REPORTING CONFIG FAILED- error code: ${descMap.data[0]}"
+			}
+		}
+	} else if (map.name == "temperature") {
+		if (tempOffset) {
+			map.value = (int) map.value + (int) tempOffset
+		}
+		map.descriptionText = temperatureScale == 'C' ? '{{ device.displayName }} was {{ value }}°C' : '{{ device.displayName }} was {{ value }}°F'
+		map.translatable = true
+	} else if (map.name == "humidity") {
+		if (humidityOffset) {
+			map.value = (int) map.value + (int) humidityOffset
+		}
+	}
+
+	log.debug "Parse returned $map"
+	return map ? createEvent(map) : [:]
+}
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	return refresh()
+}
+
+def refresh() {
+	log.debug "refresh temperature, humidity"
+	return zigbee.readAttribute(0x0402, 0x0000, [destEndpoint: 0x01])+
+		  zigbee.readAttribute(0x0405, 0x0000, [destEndpoint: 0x02])
+
+}
+
+def configure() {
+	// Device-Watch allows 2 check-in misses from device + ping (plus 1 min lag time)
+	// enrolls with default periodic reporting until newer 5 min interval is confirmed
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 1 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	log.debug "Configuring Reporting and Bindings."
+
+	// temperature minReportTime 30 seconds, maxReportTime 5 min. Reporting interval if no activity
+	return refresh() +
+		  zigbee.configureReporting(0x0405, 0x0000, DataType.UINT16, 30, 3600, 100, [destEndpoint: 0x02]) +
+                  zigbee.configureReporting(0x0402, 0x0000, DataType.UINT16, 30, 3600, 100, [destEndpoint: 0x02])
```

Comment:
should this be endpoint 1?  in the readAttr in refresh temperature is endpoint 1, but here it is endpoint 2